crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -0,0 +1,210 @@
+use api_models::webhooks;
+use diesel_models::{dispute, enums, payment_intent::PaymentIntent, refund};
+use error_stack::{IntoReport, ResultExt};
+use futures::TryFutureExt;
+
+use super::create_event_and_trigger_outgoing_webhook;
+use crate::{
+    core::{
+        errors, payments,
+        webhooks::types::{OutgoingWebhookTrigger, OutgoingWebhookType},
+    },
+    routes::AppState,
+    services,
+    types::{
+        api, domain,
+        transformers::{ForeignInto, ForeignTryInto},
+    },
+};
+
+#[async_trait::async_trait]
+impl OutgoingWebhookTrigger for PaymentIntent {
+    async fn construct_outgoing_webhook_content(
+        &self,
+        state: &AppState,
+        merchant_account: domain::MerchantAccount,
+        merchant_key_store: domain::MerchantKeyStore,
+    ) -> errors::CustomResult<webhooks::OutgoingWebhookContent, errors::ApiErrorResponse> {
+        payments::payments_core::<api::PSync, api::PaymentsResponse, _, _, _>(
+            state,
+            merchant_account,
+            merchant_key_store,
+            payments::operations::PaymentStatus,
+            api::PaymentsRetrieveRequest {
+                resource_id: api_models::payments::PaymentIdType::PaymentIntentId(
+                    self.payment_id.clone(),
+                ),
+                merchant_id: Some(self.merchant_id.clone()),
+                force_sync: false,
+                ..Default::default()
+            },
+            services::AuthFlow::Merchant,
+            payments::CallConnectorAction::Avoid,
+        )
```

Comment:
Do not call `payments_core` again, this might lead to infinite recursion in rare cases. Instead use the payment data, construct the webhook content and insert that into process tracker.